app/src/Category.js

```diff
@@ -9,11 +9,13 @@ import Filters from "./Filters";
 export default function Category(props) {
   const tree = props.tree;
   const data = props.data;
-  const k = props.k;
-  const v = props.v;
+  const id = props.id && props.id.match(/^(\w+?)\/(\w+?)\|(.+)$/);
+  const k = id ? id[1] : props.k;
+  const v = id ? id[2] : props.v;
   const kv = `${k}/${v}`;
   const entries = data.dict && data.dict[k] && data.dict[k][v];
   const hash = props.location.hash;
+  const slug = id ? id[3] : (hash && hash.slice(1)); // remove leading '#'
```

Comment:
We need to prefer the hash over the slug in the ID. In fact, when the user clicks one of the “#” links, we should erase that part of the slug in the `id` parameter.